packages/datagateway-dataview/src/views/card/entityCard.component.tsx

```diff
@@ -0,0 +1,397 @@
+
+    shadowInvisible: {
+      visibility: 'hidden',
+      opacity: 0,
+      transition: 'visibility 0s, opacity 0.5s linear',
+    },
+
+    information: {
+      display: 'flex',
+      paddingLeft: '15px',
+
+      // Apply a small space for each Typography component (p).
+      '& p': {
+        paddingTop: '5px',
+      },
+    },
+
+    informationLabel: {
+      float: 'left',
+      '& p': {
+        // Support aligning icons and label.
+        display: 'flex',
+      },
+    },
+
+    informationData: {
+      float: 'right',
+      textAlign: 'left',
+      paddingLeft: '5px',
+
+      '& p': {
+        display: 'block',
+        whiteSpace: 'nowrap',
+        maxWidth: '10vw',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      },
+    },
+
+    buttons: {
+      padding: '10px',
+      textAlign: 'center',
+      '& div': {
+        paddingTop: '10px',
+      },
+    },
+
+    moreInformation: {
+      paddingTop: '10px',
+    },
+
+    tags: {
+      paddingTop: '10px',
+    },
+
+    chip: {
+      margin: theme.spacing(0.5),
+    },
+  });
+  return styles;
+});
+
+export interface EntityImageDetails {
+  url: string;
+  title?: string;
+}
+
+export interface EntityCardDetails {
+  label: string;
+  content?: React.ReactNode;
+  icon?: JSX.Element;
+}
+
+interface EntityCardProps {
+  title: EntityCardDetails;
+
+  description?: string;
+  information?: EntityCardDetails[];
+
+  moreInformation?: React.ReactNode;
+  buttons?: React.ReactNode[];
+
+  image?: EntityImageDetails;
+  tags?: string[];
+}
+
+const EntityCard = (props: EntityCardProps): React.ReactElement => {
+  const classes = useCardStyles();
+  const {
+    title,
+    description,
+    information,
+    moreInformation,
+    buttons,
+    image,
+    tags,
+  } = props;
+
+  // The default collapsed height for card description is 100px.
+  const defaultCollapsedHeight = 100;
+  const [isDescriptionCollapsed, setDescriptionCollapsed] = React.useState(
+    false
+  );
+  const descriptionRef = React.useRef<HTMLParagraphElement>(null);
+  const [collapsibleInteraction, setCollapsibleInteraction] = React.useState(
+    false
+  );
+  const [isMoreInfoCollapsed, setMoreInfoCollapsed] = React.useState(false);
+
+  React.useEffect(() => {
+    // Decide if the collapsible should be present depending on
+    // if the description height exceeds the default collapsed height.
+    if (descriptionRef && descriptionRef.current) {
+      if (descriptionRef.current.clientHeight > defaultCollapsedHeight)
+        setCollapsibleInteraction(true);
+    }
+  }, [setCollapsibleInteraction]);
+
+  return (
+    <Card id="card" className={classes.root}>
+      {/* We allow for additional width when having an image in the card (see card styles). */}
+      {image && (
+        <CardMedia
+          aria-label="card-image"
+          component="img"
+          className={classes.cardImage}
+          image={image.url}
+          title={image.title && image.title}
```

Comment:
@patrick-austin this was just done so that we didn't have to deal with an `undefined` when it comes to rendering the title. It would just give the title in the event it is present.